projects/packages/videopress/src/client/lib/url/index.ts

```diff
@@ -129,3 +129,41 @@ export function buildVideoPressURL(
 export const removeFileNameExtension = ( name: string ) => {
 	return name.replace( /\.[^/.]+$/, '' );
 };
+
+/**
+ * Helper function to create and return textarea element.
+ * Based on https://github.com/Automattic/wp-calypso/blob/1ea156fe734d57fdf13cd332e82ac688eacd3bee/client/lib/formatting/decode/browser.js#L9
+ *
+ * > Moreover, using textContent can prevent XSS attacks.
+ *
+ * @see https://developer.mozilla.org/en-US/docs/Web/API/Node/textContent#differences_from_innerhtml
+ *
+ * It will be used to decode HTML entities,
+ * As long as element doesn’t get inserted in the DOM,
+ * we’re good in terms of security,
+ * since textContent will return the content without evaluating it.
+ * @returns {HTMLTextAreaElement} - Textarea element
+ */
+const createTextareaElement = (): HTMLTextAreaElement => {
+	if ( document.implementation && document.implementation.createHTMLDocument ) {
+		return document.implementation.createHTMLDocument( '' ).createElement( 'textarea' );
+	}
+
+	return document.createElement( 'textarea' );
+};
+
+/**
+ * Decode the given text, replacing HTML entities
+ * with their corresponding characters.
+ *
+ * @param {string} text     - Text to decode
+ * @returns {string | null} Decoded text
```

Comment:
This can no longer be a `null`, right?

Also, isn't JSDoc for a typed function redundant? Perhaps we can just get rid of it?

```suggestion
 * @returns {string} Decoded text
```